### PR TITLE
KAFKA-9823: Follow-up, check state for handling commit error response

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@ableegoldman @hachikuji

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java [hunk 1]
```diff
@@ -1231,13 +1235,13 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
 
                             // only need to reset generation and re-join group if generation has not changed;
                             // otherwise only raise rebalance-in-progress error
```
Comment (ableegoldman) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java:1237]: Should we update the comment here (and on line 1205)?

**Review by ableegoldman — APPROVED** [review]
Thanks for the quick fix!

**Comment by ableegoldman** [comment]
Flaky test failures:
`QueryableStateIntegrationTest.shouldAllowConcurrentAccesses`
`ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup`
`OptimizedKTableIntegrationTest.shouldApplyUpdatesToStandbyStore`

**Comment by guozhangwang** [comment]
test this please

**Review by vvcephei — APPROVED** [review]
Thanks @guozhangwang , as far as I understand the Consumer, this looks good to me.

**Pull request merged**